service/storage/operations_internal_test.go

```diff
@@ -1255,3 +1256,113 @@ func TestIndexAndLookup_Seals(t *testing.T) {
 		assert.Equal(t, 1, decodeCallCount)
 	})
 }
+
+func TestLibrary_IterateLedger(t *testing.T) {
+	entries := 5
+	paths := mocks.GenericLedgerPaths(entries)
+	payloads := mocks.GenericLedgerPayloads(entries)
+
+	t.Run("nominal case", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		codec := zbor.NewCodec()
+		l := &Library{codec}
+
+		for i := 0; i < entries; i++ {
+			require.NoError(t, db.Update(l.SavePayload(mocks.GenericHeight, paths[i], payloads[i])))
+		}
+
+		var callCount int
+		op := l.IterateLedger(func(path ledger.Path, payload *ledger.Payload) error {
+			callCount++
+
+			assert.Contains(t, paths, path)
+			assert.Contains(t, payloads, payload)
+
+			return nil
+		})
+
+		err := db.View(op)
+
+		assert.NoError(t, err)
+		assert.Equal(t, entries, callCount)
+	})
+
+	t.Run("handles multiple payloads with the same path", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		codec := zbor.NewCodec()
+		l := &Library{codec}
+
+		for i := 0; i < entries; i++ {
+			// Always use paths[0] for every payload.
+			require.NoError(t, db.Update(l.SavePayload(mocks.GenericHeight, paths[0], payloads[i])))
```

Comment:
This is still not testing what is important. When we have multiple payloads with the same path, the one with the highest height should be iterated over _only_. So we should use different heights here and ensure that the correct payload is iterated.